subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -75,9 +70,9 @@ static struct hawkbit_config {
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 	char server_addr[DNS_MAX_NAME_SIZE + 1];
 	char server_port[sizeof(STRINGIFY(__UINT16_MAX__))];
-#ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
-	char ddi_security_token[DDI_SECURITY_TOKEN_SIZE + 1];
-#endif
+	IF_DISABLED(CONFIG_HAWKBIT_DDI_NO_SECURITY,
+		    (char ddi_security_token[DDI_SECURITY_TOKEN_SIZE + 1];))
+	IF_ENABLED(CONFIG_HAWKBIT_USE_DYNAMIC_CERT_TAG, (sec_tag_t tls_tag;))
```

Comment:
IMHO the old implementation is more readable and generally accepted within Zephyr for struct declaration. The `IF_ENABLED`/`IF_DISABLED` is generally used inline in C code or in macros for initialization